engine/engine.go

```diff
@@ -250,43 +257,69 @@ func (e *Engine) PublishLatest(ctx context.Context) (cid.Cid, error) {
 	return adCid, nil
 }
 
-// PublishLatestHTTP publishes the latest existing advertisement to a specific
-// indexer.
-func (e *Engine) PublishLatestHTTP(ctx context.Context, indexerHost string) (cid.Cid, error) {
+// PublishLatestHTTP publishes the latest existing advertisement to the
+// specific indexers.
+func (e *Engine) PublishLatestHTTP(ctx context.Context, announceURLs ...*url.URL) (cid.Cid, error) {
 	adCid, err := e.latestAdToPublish(ctx)
 	if err != nil {
 		return cid.Undef, err
 	}
 
+	err = e.httpAnnounce(ctx, adCid, announceURLs)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return adCid, nil
+}
+
+func (e *Engine) httpAnnounce(ctx context.Context, adCid cid.Cid, announceURLs []*url.URL) error {
 	ai := &peer.AddrInfo{
 		ID: e.h.ID(),
 	}
 
 	switch e.pubKind {
 	case NoPublisher:
 		log.Info("Remote announcements disabled")
-		return cid.Undef, nil
+		return nil
 	case DataTransferPublisher:
 		ai.Addrs = e.h.Addrs()
```

Comment:
For libp2p you can get the host's addrinfo directly from host via: `h.Peerstore().PeerInfo(h.ID())`.

Reply:
Since I already have the host, it seemed more direct to call `h.Addrs()` here.  In the other case of HTTP publisher, I do not want all the addresses, just the http address of the HTTP publisher.  For those reasons, I get the addresses part of the `AddrInfo` separately instead of getting the whole `AddrInfo`.